sdk/spring-sdk/src/main/java/kalix/springsdk/valueentity/ReflectiveValueEntityProvider.java

```diff
@@ -42,22 +42,27 @@
   private final Descriptors.ServiceDescriptor serviceDescriptor;
   private final ComponentDescription componentDescription;
 
-  public static <S, E extends ValueEntity<S>> ReflectiveValueEntityProvider<S, E> of(
-      String entityType, Class<E> cls, Function<ValueEntityContext, E> factory) {
+  public static <S, E extends ValueEntity<S>> ReflectiveValueEntityProvider<S, E> of(Class<E> cls, Function<ValueEntityContext, E> factory) {
     return new ReflectiveValueEntityProvider<>(
-        entityType, cls, factory, ValueEntityOptions.defaults());
+         cls, factory, ValueEntityOptions.defaults());
   }
 
   public ReflectiveValueEntityProvider(
-      String entityType,
-      Class<E> cls,
+
+      Class<E> entityClass,
       Function<ValueEntityContext, E> factory,
       ValueEntityOptions options) {
-    this.entityType = entityType;
+
+    Entity annotation = entityClass.getAnnotation(Entity.class);
+    if (annotation == null)
+      throw new IllegalArgumentException("Value Entity [" + entityClass.getName() + "] is missing '@Entity' annotation");
```

Comment:
This validation won't be needed when we have autowire. We won't build a reflective provider if the entity is not annotated with `@Entity`. 

On the other hand, we may need to validate that `entityType` and `entityKey` are not empty string, but this can be validated when wiring things up and not on each provider.

Reply:
For the API-first services we don't require entityType but synthesise it if not specified, so I guess we need to deal with that as well if we don't already,

Reply:
Maybe I misremembered and we always require it there as well.

Reply:
We do require it. We can't create one based on class name because users may change the class name and then they lose the connection to the existing journal.